pkg/reconciler/cluster/cluster.go

```diff
@@ -123,14 +132,48 @@ func (c *Controller) reconcile(ctx context.Context, cluster *v1alpha1.Cluster) e
 			cluster.Status.Conditions.SetReady(corev1.ConditionUnknown,
 				"SyncerInstalling",
 				"Installing syncer on cluster")
-		} else {
+		case SyncerModePush:
+			// TODO: specify more types
+			syncedResourceTypes := []string{"pods", "deployments"}
+
+			kubeConfig.CurrentContext = logicalCluster
+			bytes, err := clientcmd.Write(*kubeConfig)
+			if err != nil {
+				log.Printf("error writing kubeconfig for syncer: %v", err)
+				cluster.Status.Conditions.SetReady(corev1.ConditionFalse,
+					"ErrorInstallingSyncer",
+					fmt.Sprintf("Error installing syncer: %v", err))
+				return nil // Don't retry.
+			}
+			from, err := clientcmd.RESTConfigFromKubeConfig(bytes)
```

Comment:
Instead of writing the `kubeconfig` and parsing it again, you should be able to use:

```suggestion
			from, err := clientcmd.NewNonInteractiveClientConfig(*kubeConfig, logicalCluster, &clientcmd.ConfigOverrides{}, nil).ClientConfig()
```

Reply:
Very nice, done.